Packs/TrustwaveFusion/Integrations/TrustwaveFusion/TrustwaveFusion.py

```diff
@@ -0,0 +1,587 @@
+    demisto.debug(f"last_run: {last_run}")
+    if last_fetch is None:
+        # if missing, use what provided via first_fetch
+        last_fetch = first_fetch
+    else:
+        # otherwise use the stored last fetch
+        last_fetch = int(last_fetch)
+
+    search_since = max(first_fetch, last_fetch)
+    demisto.debug(f"{first_fetch} <=> {last_fetch} == {search_since}")
+    created_since = timestamp_to_datestring(search_since * 1000, FUSION_DATE_FORMAT)
+
+    ticket_types = demisto.params().get("ticket_types")
+    incidents = []
+    try:
+        params = {
+            "pageSize": max_results,
+            "createdSince": created_since,
+            "sortField": "createdOn",
+            "sortDescending": "false",
+        }
+        if ticket_types:
+            params["type"] = ",".join(ticket_types)
+        demisto.debug(f"params: {params}")
+        tickets = client.search_tickets(**params)
+        latest_timestamp = search_since
+        if tickets:
+            demisto.debug(f"Found {len(tickets)} tickets from Fusion")
+            for tkt in tickets:
+                incident = {
+                    "name": tkt["subject"],
+                    "occurred": tkt["createdOn"],
+                    "description": tkt.get("description", ""),
+                    "severity": SEVERITY_MAP.get(tkt.get("priority"), 0),
+                    "rawJSON": json.dumps(tkt),
+                }
+                dt = dateparser.parse(tkt["createdOn"], settings={"TIMEZONE": "UTC"})
+                latest_timestamp = max(dt.timestamp(), latest_timestamp)
+                incidents.append(incident)
+    except DemistoException as e:
+        demisto.error(f"ERROR: {e}")
+        demisto.error(e.res)
+
+    # One second in the future to prevent duplicates
+    last_run = {"last_fetch": latest_timestamp + 1}
+    demisto.setLastRun(last_run)
+    demisto.incidents(incidents)
+
+
+def get_ticket_command(client, args):
+    ticket = client.get_ticket(args.get("id"))
+
+    readable = [
+        "| field | value |",
+        "|-|-|",
+    ]
+    for k, v in ticket.items():
+        str_value = stringEscapeMD(formatCell(v), True, True)
+        if len(str_value) > 200:
+            str_value = str_value[:200] + "....[Trucated]"
+        readable.append(f"| {k} | {str_value} |")
+
+    command_results = CommandResults(
+        outputs_prefix="Trustwave.Ticket",
+        outputs_key_field="number",
+        outputs=ticket,
+        readable_output="\n".join(readable),
+    )
+    return command_results
+
+
+def add_ticket_comment_command(client, args):
+    id = args.get("id")
+    comment = args.get("comment")
+    client.add_ticket_comment(id, comment)
+
+    return "Success"
+
+
+def close_ticket_command(client, args):
+    id = args.get("id")
+    comment = args.get("comment")
+    client.add_ticket_comment(id, comment)
+
+    return "Success"
+
+
+def get_finding_command(client, args):
+    finding = client.get_finding(args.get("id"))
```

Comment:
the same as getting a ticket, if the finding with the given id is not found, return a readable output that states so and not an error.